File: SportsSideKick/app/src/main/java/base/app/util/ui/NextMatchView.java
package base.app.util.ui;

import android.content.Context;
import android.os.Handler;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.bumptech.glide.Glide;

import base.app.R;
import base.app.data.ticker.NewsTickerInfo;
import base.app.data.ticker.NextMatchModel;
import base.app.util.commons.NextMatchCountdown;
import butterknife.BindView;
import butterknife.ButterKnife;


/**
 * Created by Filip on 10/23/2017.
 * Copyright by Hypercube d.o.o.
 * www.hypercubesoft.com
 */

public class NextMatchView extends RelativeLayout {

    @BindView(R.id.next_match_container)
    View nextMatchContainer;

    @BindView(R.id.left_image)
    ImageView logoOfFirstTeam;

    @BindView(R.id.right_image)
    ImageView logoOfSecondTeam;

    @BindView(R.id.left_name)
    TextView nameOfFirstTeam;

    @BindView(R.id.right_name)
    TextView nameOfSecondTeam;

    @BindView(R.id.date)
    TextView dateTextView;

    @BindView(R.id.countdown)
    TextView countdownTextView;

    @BindView(R.id.background)
    ImageView background;

    long timestamp;

    public NextMatchView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initView();
    }

    private void initView() {
        View view = inflate(getContext(), R.layout.next_match_view, this);
        ButterKnife.bind(view);

        NewsTickerInfo info = NextMatchModel.getInstance().loadTickerInfoFromCache();
        if(info!=null && NextMatchModel.getInstance().isNextMatchUpcoming()){
            updateCountdownTimer();
            if(logoOfFirstTeam.getDrawable()==null){
                Glide.with(this).load(info.getFirstClubUrl()).into(logoOfFirstTeam);
                Glide.with(this).load(info.getSecondClubUrl()).into(logoOfSecondTeam);
            }
            timestamp = Long.parseLong(info.getMatchDate());
            nameOfFirstTeam.setText(info.getFirstClubName());
            nameOfSecondTeam.setText(info.getSecondClubName());
            dateTextView.setText(NextMatchCountdown.getTextValue(getContext(),timestamp,true));

            final Handler handler = new Handler();
            final int delay = 100; //milliseconds

            handler.postDelayed(new Runnable(){
                public void run(){
                    updateCountdownTimer();
                    handler.postDelayed(this, delay);
                }
            }, delay);
            nextMatchContainer.setVisibility(View.VISIBLE);

            Glide.with(getContext())
                    .load(R.drawable.video_chat_background)
                    .into(background);
        } else {
            nextMatchContainer.setVisibility(View.GONE);
        }
    }

    private void updateCountdownTimer(){
        countdownTextView.setText(NextMatchCountdown.getCountdownValue(timestamp));
    }
}

File: SportsSideKick/app/src/main/java/base/app/util/ui/NextMatchView.kt
package base.app.util.ui

import android.content.Context
import android.os.Build.VERSION
import android.os.Build.VERSION_CODES
import android.text.Html
import android.text.Html.FROM_HTML_MODE_LEGACY
import android.util.AttributeSet
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import base.app.R
import base.app.R.string
import base.app.data.wall.ticker.NextMatchModel
import base.app.util.commons.NextMatchCountdown
import io.reactivex.Observable
import io.reactivex.android.schedulers.AndroidSchedulers.mainThread
import io.reactivex.disposables.CompositeDisposable
import kotlinx.android.synthetic.main.next_match_view.view.*
import java.lang.Long.parseLong
import java.util.concurrent.TimeUnit

/**
 * Created by Filip on 10/23/2017.
 * Copyright by Hypercube d.o.o.
 * www.hypercubesoft.com
 */
class NextMatchView(context: Context, attrs: AttributeSet)
    : RelativeLayout(context, attrs) {

    private val disposables = CompositeDisposable()

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        showMatchInfo()
    }

    override fun onDetachedFromWindow() {
        disposables.clear()
        super.onDetachedFromWindow()
    }

    private fun showMatchInfo() {
        inflate(R.layout.next_match_view)

        val info = NextMatchModel.getInstance().loadTickerInfoFromCache()
        if (info != null && NextMatchModel.getInstance().isNextMatchUpcoming) {
            logoOfFirstTeam.show(info.firstClubUrl)
            logoOfSecondTeam.show(R.drawable.club_logo)

            nameOfFirstTeam.text = info.firstClubName
            nameOfSecondTeam.text = info.secondClubName

            val timestamp = parseLong(info.matchDate)
            val textValue = NextMatchCountdown.getTextValue(context, timestamp, true)
            date.text = textValue

            disposables.add(Observable.interval(
                    1000,
                    500,
                    TimeUnit.MILLISECONDS)
                    .observeOn(mainThread())
                    .subscribe {
                        if (textValue == context.getString(R.string.live)) {
                            countdown.text = date.text
                        } else {
                            updateCountdownTimer(timestamp)
                        }
                    })

            nextMatchSplashContainer.visible()
        } else {
            nextMatchSplashContainer.hide()
        }
        showSloganText()
    }

    private fun showSloganText() {
        if (VERSION.SDK_INT >= VERSION_CODES.N) {
            label.text = Html.fromHtml(context.getString(string.slogan), FROM_HTML_MODE_LEGACY)
        } else {
            @SuppressWarnings("deprecation")
            label.text = Html.fromHtml(context.getString(string.slogan))
        }
    }

    private fun updateCountdownTimer(timestamp: Long) {
        countdown.text = NextMatchCountdown.getCountdownValue(timestamp)
    }

    fun ViewGroup.inflate(layoutId: Int) =
            View.inflate(context, layoutId, this)
}
